client/src/main/scala/drt/client/components/DateSelector.scala

```diff
@@ -0,0 +1,72 @@
+package drt.client.components
+
+import drt.client.logger.{Logger, LoggerFactory}
+import drt.client.services.JSDateConversions.SDate
+import drt.shared.SDateLike
+import japgolly.scalajs.react.extra.Reusability
+import japgolly.scalajs.react.vdom.TagOf
+import japgolly.scalajs.react.vdom.html_<^._
+import japgolly.scalajs.react.{Callback, CallbackTo, ReactEventFromInput, ScalaComponent, _}
+import org.scalajs.dom.html.Div
+
+import scala.scalajs.js.Date
+
+object DateSelector {
+  val log: Logger = LoggerFactory.getLogger(getClass.getName)
+
+  case class Props(label: String, selectedDate: SDateLike, callback: SDateLike => Callback)
+
+  case class State(day: Int, month: Int, year: Int) {
+    def selectedDate = SDate(year, month, day)
+  }
+
+  val today: SDateLike = SDate.now()
+
+  def daysInMonth(month: Int, year: Int) = new Date(year, month, 0).getDate()
+
+  def formRow(label: String, xs: TagMod*): TagOf[Div] = {
+    <.div(^.className := "form-group row",
+      <.label(label, ^.className := "col-sm-1 col-form-label"),
+      <.div(^.className := "col-sm-8", xs.toTagMod))
+  }
+
+  implicit val propsReuse: Reusability[Props] = Reusability.by(p => p.selectedDate.toISOString())
+  implicit val stateReuse: Reusability[State] = Reusability.derive[State]
+
+  val component = ScalaComponent.builder[Props]("DatePicker")
+    .initialStateFromProps(
+      p => {
+        State(day = p.selectedDate.getDate(), month = p.selectedDate.getMonth(), year = p.selectedDate.getFullYear())
+      }
+    )
+    .renderPS(r = (scope, props, state) => {
+      val months = Seq("January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December").zip(1 to 12)
+      val days = Seq.range(1, 32)
```

Comment:
32 days in a month?

Reply:
range doesn't include the last number, which I agree looks a bit confusing.